Doesn't build with Boost 1.71 (Ubuntu Focal)

Issue is here:

```
.../src/sick_safetyscanners/src/communication/AsyncTCPClient.cpp:51:91:   required from here
/usr/include/boost/asio/detail/io_object_impl.hpp:87:40: error: const class boost::reference_wrapper<boost::asio::io_context> has no member named get_executor
```

And the same thing in `AsyncUDPClient.cpp`.

Related: https://github.com/boostorg/asio/issues/227

--- a/src/communication/AsyncTCPClient.cpp
+++ b/src/communication/AsyncTCPClient.cpp
@@ -45,10 +45,10 @@ AsyncTCPClient::AsyncTCPClient(const PacketHandler& packet_handler,
 
 {
   // Keep io_service busy
-  m_io_work_ptr = std::make_shared<boost::asio::io_service::work>(boost::ref(m_io_service));
+  m_io_work_ptr = std::make_shared<boost::asio::io_service::work>(m_io_service);
   try
   {
-    m_socket_ptr = std::make_shared<boost::asio::ip::tcp::socket>(boost::ref(m_io_service));
+    m_socket_ptr = std::make_shared<boost::asio::ip::tcp::socket>(m_io_service);
   }
   catch (const std::exception& e)
   {

--- a/src/communication/AsyncUDPClient.cpp
+++ b/src/communication/AsyncUDPClient.cpp
@@ -44,12 +44,11 @@ AsyncUDPClient::AsyncUDPClient(const PacketHandler& packet_handler,
   , m_io_service(io_service)
 {
   // Keep io_service busy
-  m_io_work_ptr = std::make_shared<boost::asio::io_service::work>(boost::ref(m_io_service));
+  m_io_work_ptr = std::make_shared<boost::asio::io_service::work>(m_io_service);
   try
   {
-    m_socket_ptr = std::make_shared<boost::asio::ip::udp::socket>(
-      boost::ref(m_io_service),
-      boost::asio::ip::udp::endpoint(boost::asio::ip::udp::v4(), local_port));
+    auto endpoint = boost::asio::ip::udp::endpoint(boost::asio::ip::udp::v4(), local_port);
+    m_socket_ptr = std::make_shared<boost::asio::ip::udp::socket>(m_io_service, endpoint);
   }
   catch (const std::exception& e)
   {

